app/scripts/migrations/148.ts

```diff
@@ -0,0 +1,87 @@
+import { toHex } from '@metamask/controller-utils';
+import { getErrorMessage, hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+export const version = 148;
+
+/**
+ * This migration resets sessionData and isSignedIn if using the old sessionData state shape.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  try {
+    versionedData.data = transformState(versionedData.data);
+  } catch (error) {
+    global.sentry?.captureException?.(
+      new Error(`Migration #${version}: ${getErrorMessage(error)}`),
+    );
+  }
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  const newState = cloneDeep(state);
+
+  if (!hasProperty(newState, 'NetworkController')) {
+    throw new Error(`newState.NetworkController must be present`);
```

Comment:
did we want to throw or log to sentry and then return the old state?

Reply:
I added a try/catch to the `migrate` function that automatically logs to Sentry and returns the old state if there is an error in `transformState`. This is a bit of a different pattern than is in other migration files, but it seemed less error-prone. I can make the error handling the same if you like though.

Reply:
Wasn't aware that that way the behavior. What you have is definitely cleaner. Please ignore my comment